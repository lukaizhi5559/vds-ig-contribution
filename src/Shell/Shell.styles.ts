/*!
 * Copyright (C) Verizon. All rights reserved.
 */

import { createUseStyles } from 'react-jss';

/**
 * Styles for the Shell slots
 */
export const useShellStyles = createUseStyles({
  root: {
    backgroundColor: '#f9f9f9',
    minHeight: '100vh',
    minWidth: '80vw',
    display: 'flex',
    flexDirection: 'column',
  },
  header: {
    backgroundColor: 'white',
    boxShadow: '0 1px 3px rgba(0, 0, 0, 0.1)',
  },
  container: {
    maxWidth: '1240px',
    margin: '0 auto',
    padding: '1rem 1.5rem',
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  nav: {
    display: 'flex',
    marginLeft: '2rem',
    '& a': {
      color: '#4a4a4a',
      marginLeft: '1.5rem',
      textDecoration: 'none',
      '&:hover': {
        color: 'black',
      },
    },
  },
  logo: {
    fontSize: '1.25rem',
    fontWeight: 'bold',
  },
  searchIcon: {
    backgroundColor: '#b0b0b0',
    borderRadius: '50%',
    height: '2rem',
    width: '2rem',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  },
  main: {
    margin: '0 auto',
    padding: '2.5rem 1.5rem',
    display: 'grid',
  },
  footer: {
    backgroundColor: '#f1f1f1',
    padding: '1rem 1.5rem',
    marginTop: 'auto',
    textAlign: 'center',
  },
  footerLink: {
    marginRight: '2rem',    
  }
});