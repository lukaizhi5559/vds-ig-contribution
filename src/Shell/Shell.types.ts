
/*!
 * Copyright (C) Verizon. All rights reserved.
 */

import * as React from "react";

/**
 *  ShellProps
 */
export type ShellProps = React.PropsWithChildren & {
};
