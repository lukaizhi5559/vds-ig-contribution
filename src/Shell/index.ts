
/*!
 * Copyright (C) Verizon. All rights reserved.
 */

export { Shell } from './Shell';
export type { ShellProps } from './Shell.types';
