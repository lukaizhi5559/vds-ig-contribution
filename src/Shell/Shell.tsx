/*!
 * Copyright (C) Verizon. All rights reserved.
 */

import React, { useEffect, useRef, useState } from "react";
import { Outlet } from "@tanstack/react-router";
import { useNavigate } from "@tanstack/react-router";
import { useUserContext } from "@/context/UserContext";
import { getInitials } from "@/lib/utils";
import { useShellStyles } from "./Shell.styles";
import type { ShellProps } from "./Shell.types";

/**
 * Render the final JSX of Shell
 */
export const Shell: React.FC<ShellProps> = () => {
  const styles = useShellStyles();
  const navigate = useNavigate();
  const dropdownRef = useRef<HTMLDivElement>(null);

  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
  const { userDetails, isAuthenticated, logout } = useUserContext();

  useEffect(() => {
    if (!isAuthenticated) {
      navigate({ to: "/login", replace: true });
    }
  }, [isAuthenticated, navigate]);

  useEffect(() => {
    const handleClickOutside = (event: MouseEvent) => {
      if (
        dropdownRef.current &&
        !dropdownRef.current.contains(event.target as Node)
      ) {
        setIsDropdownOpen(false);
      }
    };

    document.addEventListener("mousedown", handleClickOutside);
    return () => {
      document.removeEventListener("mousedown", handleClickOutside);
    };
  }, []);

  return (
    <div className={styles.root}>
      {/* Header */}
      <header className={styles.header}>
        <div className={styles.container}>
          <div className="flex items-center">
            <a href="/" className={styles.logo}>
              Verizon
            </a>
            <nav className={styles.nav}>
              {/* Uncomment possible nav links */}
              {/* <a href="#">Mobile</a>
              <a href="#">Internet</a>
              <a href="#">Solutions</a>
              <a href="#">Insights</a> */}
            </nav>
          </div>
          <div className={styles.userSection}>
            {userDetails && (
              <span className={styles.userName}>
                Welcome, {userDetails.name}
              </span>
            )}
            <div
              className={styles.dropdownContainer}
              ref={dropdownRef}
            >
              <button
                className={styles.searchIcon}
                onClick={() => setIsDropdownOpen(!isDropdownOpen)}
              >
                {getInitials(userDetails?.name)}
              </button>
              {isDropdownOpen && (
                <div className={styles.dropdown}>
                  <a
                    className={styles.dropdownLink}
                    href="/"
                  >
                    Home
                  </a>
                  <a
                    className={styles.dropdownLink}
                    href="/submission-detail"
                  >
                    Submission Detail
                  </a>
                  <button
                    className={styles.dropdownLink}
                    onClick={() => {
                      setIsDropdownOpen(false);
                      logout();
                    }}
                  >
                    Logout
                  </button>
                </div>
              )}
            </div>
          </div>
        </div>
      </header>

      {/* Portal Page Content */}
      <main className={styles.main}>
        <Outlet /> {/* Child routes will render here */}
      </main>

      {/* Footer */}
      <footer className={styles.footer}>
        <div>
          <a href="#" className={styles.footerLink}>
            Privacy Policy
          </a>
          <a href="#" className={styles.footerLink}>
            Terms of Service
          </a>
          <a href="#">Contact Us</a>
        </div>
      </footer>
    </div>
  );
};
