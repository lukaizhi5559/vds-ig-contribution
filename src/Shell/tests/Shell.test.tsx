
/*!
 * Copyright (C) Verizon. All rights reserved.
 */

import * as React from 'react';
import { render } from '@testing-library/react';
import { axe, toHaveNoViolations } from 'jest-axe';

import { Shell } from '../Shell';

expect.extend(toHaveNoViolations);

describe('Shell', () => {
  it('should not have a11y violations', async () => {
    const { container } = render(<Shell>Default Shell A11y</Shell>);
    expect(await axe(container)).toHaveNoViolations();
  });

  it('renders a default state', () => {
    const { container } = render(<Shell>Default Shell</Shell>);
    expect(container).toMatchSnapshot();
  });
});
