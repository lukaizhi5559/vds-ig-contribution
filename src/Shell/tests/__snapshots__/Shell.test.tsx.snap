// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shell renders a default state 1`] = `
<div>
  <div
    class="root-0-2-1"
  >
    READY
  </div>
</div>
`;
